reliability/single-fit/rum/src/test/kotlin/com/datadog/android/rum/integration/ManualTrackingRumTest.kt

```diff
@@ -395,6 +400,217 @@ class ManualTrackingRumTest {
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewUrl(key)
+                hasViewName(name)
+                hasActionCount(0)
+            }
+            .hasRumEvent(index = 1) {
+                // view updated with loading time
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewUrl(key)
+                hasViewName(name)
+                hasViewLoadingTime(
+                    expectedViewLoadingTime,
+                    offset = Offset.offset(TimeUnit.MILLISECONDS.toNanos(5))
+                )
+            }
+    }
+
+    @OptIn(ExperimentalRumApi::class)
+    @Test
+    fun `M not attach view loading time W addViewLoadingTime { view was stopped }`(
+        @StringForgery key: String,
+        @StringForgery name: String,
+        @BoolForgery overwrite: Boolean
+    ) {
+        // Given
+        val rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+        rumMonitor.startView(key, name, emptyMap())
+        rumMonitor.stopView(key, emptyMap())
+
+        // When
+        rumMonitor.addViewLoadingTime(overwrite)
+
+        // Then
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.RUM_FEATURE_NAME)
+        StubEventsAssert.assertThat(eventsWritten)
+            .hasSize(2)
+            .hasRumEvent(index = 0) {
+                // Initial view
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewUrl(key)
+                hasViewName(name)
+                hasActionCount(0)
+            }
+            .hasRumEvent(index = 1) {
+                // view stopped
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewUrl(key)
+                hasViewName(name)
+                doesNotHaveViewLoadingTime()
+            }
+    }
+
+    @OptIn(ExperimentalRumApi::class)
+    @Test
+    fun `M renew view loading time W addViewLoadingTime { loading time was already added, overwrite is true }`(
+        @StringForgery key: String,
+        @StringForgery name: String,
+        @BoolForgery overwrite: Boolean
+    ) {
+        // Given
+        val rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+        val startTime = System.nanoTime()
+        rumMonitor.startView(key, name, emptyMap())
+        val intermediateTime = System.nanoTime()
+        rumMonitor.addViewLoadingTime(overwrite)
+
+        // When
+        Thread.sleep(100)
```

Comment:
why we need that?

Reply:
to simulate some time between `addViewLoadingTime` calls

Reply:
but there is no throttling mechanism here or something like that, how does it help?